int64 index
---
bool saved